src/lib/mina_generators/parties_generators.ml

```diff
@@ -244,21 +256,36 @@ let gen_account_precondition_from ?failure ~account_id ~ledger () =
             "gen_account_precondition_from: could not find account with known \
              location"
       | Some account ->
-          gen_account_precondition_from_account ?failure account )
+          gen_account_precondition_from_account ?failure account
+            ~account_state_tbl )
 
-let gen_fee (account : Account.t) =
+let gen_fee ~account_state_tbl (account : Account.t) =
+  let balance =
```

Comment:
Does it make sense to override `account` here, and then use `account.balance`? It seems like it's never useful to inspect the old account if we have an updated one.

Reply:
It is useful in some cases when account_state_tbl is optional but probably not here and in similar cases